test/suites/route-response-duplicate.spec.ts

```diff
@@ -0,0 +1,134 @@
+    const originalBodyText = await tests.app.client.getText(
+      routeResponseAceContentSelector
+    );
+    await tests.helpers.duplicateRouteResponse();
+
+    const duplicatedBodyText = await tests.app.client.getText(
+      routeResponseAceContentSelector
+    );
+    expect(duplicatedBodyText).to.equal(originalBodyText);
+  });
+
+  it('Verify duplicated route response has the same headers with the original response', async () => {
+    await tests.helpers.selectRouteResponse(1);
+    await tests.helpers.removeRouteResponse();
+    await tests.helpers.switchTab('HEADERS');
+    await tests.helpers.addHeader(
+      'route-response-headers',
+
+      { key: 'test', value: 'test' }
+    );
+    await tests.helpers.duplicateRouteResponse();
+    await tests.helpers.switchTab('HEADERS');
+
+    await tests.app.client
+      .elements('#route-response-headers .row.headers-list')
+      .should.eventually.have.property('value')
+      .to.be.an('Array')
+      .that.have.lengthOf(2);
+
+    const selectorAndValueAssertionPairs = {
+      '#route-response-headers .row.headers-list:first-child .form-control:first-child':
+        'Content-Type',
+      '#route-response-headers .row.headers-list:first-child .form-control:nth-child(2)':
+        'application/json',
+      '#route-response-headers .row.headers-list:nth-child(2) .form-control:first-child':
+        'test',
+      '#route-response-headers .row.headers-list:nth-child(2) .form-control:nth-child(2)':
+        'test'
+    };
+
+    Object.keys(selectorAndValueAssertionPairs).forEach((selector: string) => {
+      const valueToCompare = selectorAndValueAssertionPairs[selector];
+      tests.helpers.assertElementValue(selector, valueToCompare);
+    });
+  });
+
+  it('Verify duplicated route response has the same rules with the original response', async () => {
+    await tests.helpers.selectRouteResponse(1);
+    await tests.helpers.removeRouteResponse();
+    await tests.helpers.switchTab('RULES');
+    await tests.helpers.addResponseRule({
+      target: 'body',
+      isRegex: false,
+      modifier: 'test',
+      value: 'test'
+    });
+    await tests.app.client.pause(100);
+    await tests.helpers.duplicateRouteResponse();
+    await tests.helpers.switchTab('RULES');
+
+    await tests.app.client
+      .elements('app-route-response-rules .row')
+      .should.eventually.have.property('value')
+      .to.be.an('Array')
+      .that.have.lengthOf(1);
+
+    const selectorAndValueAssertionPairs = {
+      'app-route-response-rules .row select[formcontrolname="target"]': 'body',
+      'app-route-response-rules .row input[formcontrolname="modifier"]': 'test',
+      'app-route-response-rules .row input[formcontrolname="value"]': 'test'
+    };
+
+    Object.keys(selectorAndValueAssertionPairs).forEach((selector: string) => {
+      const valueToCompare = selectorAndValueAssertionPairs[selector];
+      tests.helpers.assertElementValue(selector, valueToCompare);
+    });
```

Comment:
This code cannot work. If you test with any wrong value you will see that the test is still passing. There is no await on the helper on line 99. But you cannot just put an async on the `forEach` callback. What you need to do is to put a complete `it('', async () =>{])` inside the forEach. 
You can also see a warning saying "Unhandled promise rejection warning" which means that the test failed, but the failure wasn't caught (because the test did not wait).

Reply:
Forgetting that `await` is bad on my side :) Instead of calling it four times or declaring `it` blocks with each `forEach` I've used `for...of` here. Did the work. Is that okay?